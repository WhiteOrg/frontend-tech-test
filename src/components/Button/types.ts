import { TTheme } from "../../theme/types";

type TButton = {
  color?: string;
  theme?: TTheme;
};

export type { TButton };
