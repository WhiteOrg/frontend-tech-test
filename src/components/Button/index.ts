import styled from "styled-components";
import { TButton } from "./types";

const StyledButton = styled.button<TButton>`
  background-color: ${({ theme, color = "yellow" }) => theme.colors[color]};
  min-height: 48px;
  border: none;
  text-transform: uppercase;
  font-size: 18px;
  font-weight: 600;
  letter-spacing: 2.5px;
  width: 300px;
  cursor: pointer;

  @media (max-width: 375px) {
    width: 100%;
  }
`;

export { StyledButton as Button };
