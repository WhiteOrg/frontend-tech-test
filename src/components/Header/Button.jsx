import React from 'react'
import styled from 'styled-components'


const CustomButton = styled.button`
    display: block;
    position: relative;
    width: 100%;
    max-width: 350px;
    background-color: #f9cb11;
    border: none;
    padding: 1em 0;
    color:#280055;
    font-size: 1em;
    font-weight: 550;
    text-transform: uppercase;
    letter-spacing: 2px;
    margin-bottom: 1em;
    outline-width: 0;
    transition: background-color .3s ease;
    &:hover {
        color: #000;
        cursor: pointer;
        transform: scale(1.02);
        background-color: #fff;
        border-radius: 10px;
    }
`

export default function Button({ children }) {
    return(
        <CustomButton data-testid="button-1">{children}</CustomButton>
    )

}


