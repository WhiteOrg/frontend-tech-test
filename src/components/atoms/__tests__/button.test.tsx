import React from "react";
import renderer from "react-test-renderer";
import Button from "../button";

describe("ATOMS: Render testing", () => {
  test("Should render Button", () => {
    const component = renderer.create(
      <div>
        <Button>Join</Button>
      </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
