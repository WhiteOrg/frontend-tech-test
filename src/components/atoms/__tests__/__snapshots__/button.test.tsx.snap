// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATOMS: Render testing Should render Button 1`] = `
<div>
  <button
    className="sc-AxjAm dddBUv"
  >
    Join
  </button>
</div>
`;
